manifests/init.pp

```diff
@@ -1,43 +1,42 @@
-# -- Class systemd
 # This module allows triggering systemd commands once for all modules
+#
+# @api public
+#
+# @param service_limits
+#   May be passed a resource hash suitable for passing directly into the
+#   ``create_resources()`` function as called on ``systemd::service_limits``
+#
+# @param manage_resolved
+#   Manage the systemd resolver
+#
+# @param resolved_ensure
+#   The state that the ``resolved`` service should be in
+#
+# @param manage_networkd
+#   Manage the systemd network daemon
+#
+# @param networkd_ensure
+#   The state that the ``networkd`` service should be in
+#
 class systemd (
-  $service_limits          = {},
-  Boolean $manage_resolved = true,
-  Boolean $manage_networkd = true,
+  Optional[Hash]                             $service_limits  = undef,
+  Boolean                                    $manage_resolved = false,
+  Variant[Enum['stopped','running'],Boolean] $resolved_ensure  = 'running',
+  Boolean                                    $manage_networkd = false,
+  Variant[Enum['stopped','running'],Boolean] $networkd_ensure = 'running',
```

Comment:
on the vox pupuli modules we agreed to only allow 'stopped' and 'running'. we dropped the boolean during the migration from puppet3 to puppet4/5.